pkg/types/defaults/machinepools.go

```diff
@@ -11,7 +12,10 @@ func SetMachinePoolDefaults(p *types.MachinePool, platform string) {
 	defaultReplicaCount := int64(3)
 	if platform == libvirt.Name {
 		defaultReplicaCount = 1
+	} else if platform == powervs.Name {
+		p.Architecture = "ppc64le"
```

Comment:
isn't the architecture specified explicitly in the control plane and compute. PowerVS only makes sense for ppc64le so there should be a validation failure if the user chooses something other than it.

Reply:
yes that's true. but the idea was that the defaultarch would be detected based on payload: https://github.com/openshift/installer/commit/729882a8a2c208a815d35a1923c92d0d0225852b. so if you use an openshift-install binary corresponding to P and create an install config with that, the arch would be set to ppc64le for you irrespective of the platform you are installing on (because the payload was built on a P host). in that sense the "default" when building your own installer will be based on the host system on which you build the installer. the setting of this arch based on platform conflicts with that idea.

Reply:
so, there's going to be a specific installer binary that is an amd/x86 binary, but "corresponding to P"? (b/c clearly i'm not running a binary on my laptop to do an install that was built on a power box)

or -- there needs to be doc stating that if you're installing from a different arch than you're installing onto, you need to manually edit your install-config?

Reply:
exactly so for AWS on ARM : https://console.redhat.com/openshift/install/aws/arm we have an option to select an openshift-install binary which is an x86 binary but which is built for an ARM payload, hence the install-config will  have the arch set to `arm64` when created through this binary. we would need to have the same for PowerVS too. cc @yselkowitz .

Reply:
and if you locally build the installer and use that install binary, you would have to manually edit the architecture. So, the tl;dr is - arch is based on payload. If you use an openhsift-install binary from the payload you shouldn't worry about the arch. it will be set correctly. if you build the install binary locally, you will have to change the arch manually (i.e, if you are not building on a host corresponding to the arch of the platform)

Reply:
As of 4.9 all payloads regardless of architecture have a linux/amd64 installer in addition to linux/"native".  Also, when you get the installer from the payload, it will default to the payload architecture, regardless of the installer binary runtime arch.  If you build the installer yourself, you can mimic this by running e.g. `DEFAULT_ARCH=ppc64le ./hack/build.sh`